Assets/2_MainScene/S2_Scripts/S2_EnemyScripts/S2_EnemyManager.cs

```diff
@@ -0,0 +1,360 @@
+                {
+                        if (RNG > minion_M[choice] + speeder_M[choice] + tank_M[choice])
+                            SpawnEnemy(3);
+                        else
+                            SpawnEnemy(2);
+                }
+                else
+                {
+                    int x = Random.Range(0, 100);
+                    if (x <= 50)
+                        SpawnEnemy(0);
+                    else
+                        SpawnEnemy(1);
+                }
+                break;
+            case "Hard":
+                if (RNG > speeder_H[choice])
+                {
+
+                    if (RNG > speeder_H[choice] + minion_H[choice] + tank_H[choice])
+                        SpawnEnemy(3);
+                    else
+                    {
+                        if (difficultyScale <= -2)
+                        {
+                            int x = Random.Range(0, 100);
+                            if (x < 50) SpawnEnemy(0);
+                            else SpawnEnemy(2);
+                        }
+                        else
+                        {
+                            if (RNG > speeder_H[choice] + minion_H[choice])
+                                SpawnEnemy(2);
+                            else
+                                SpawnEnemy(0);
+                        }    
+                    }
+                }
+                else
+                    SpawnEnemy(1);
+                break;
+            case "Very Hard":
+                if (RNG > minion_VH[choice])
+                {
+                    if (RNG > (minion_VH[choice] + speeder_VH[choice]))
+                    {
+
+                        if (RNG > minion_VH[choice] + speeder_VH[choice] + tank_VH[choice])
+                        {
+                                SpawnEnemy(3);
+                        }
+                        else
+                            SpawnEnemy(2);
+                    }
+                    else
+                        SpawnEnemy(1);
+                }
+                else
+                    SpawnEnemy(0);
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void RemoveFromEnemiesInWave(S2_EnemyStats x)
+    {
+        enemiesInWave.Remove(x);
+    }
+
+
+    void Start()
+    {
+        Instance = this;
+        for (int i = 0; i < enemies.Count; i++)
+        {
+            enemies[i].gameObject.SetActive(false);
+        }
+        Invoke(nameof(SpawnWave), 10);
+        GetPlayerShieldsAndHealth();
+    }
+
+    void UpgradeEnemies()
+    {
+        if (!waveComplete)
+        {
+            if (numberOfLiveEnemies >= enemyCount)
+            {
+                for (int i = 0; i < numberOfLiveEnemies; i++)
+                {
+                    enemiesInWave[i].Upgrade();
+                }
+                Invoke(nameof(UpgradeEnemies), upgradeTime);
+                print("Enemies upgraded");
```

Comment:
Console actions should be removed after it's not needed

Reply:
there is currently no player feedback associated with this behavior, will keep console until this is resolved